src/Compilers/CSharp/Test/Syntax/Parsing/SwitchExpressionParsingTests.cs

```diff
@@ -0,0 +1,917 @@
+                {
+                    N(SyntaxKind.NumericLiteralToken, "2");
+                }
+            }
+            M(SyntaxKind.CommaToken);
+            N(SyntaxKind.CloseBraceToken);
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void TestErrantCaseInSwitchExpression2()
+    {
+        UsingExpression("""
+            x switch
+            {
+                case 0: 1,
+                case 1: 2,
+            }
+            """,
+            // (2,2): error CS1041: Identifier expected; 'case' is a keyword
+            // {
+            Diagnostic(ErrorCode.ERR_IdentifierExpectedKW, "").WithArguments("", "case").WithLocation(2, 2),
+            // (3,11): error CS1003: Syntax error, '=>' expected
+            //     case 0: 1,
+            Diagnostic(ErrorCode.ERR_SyntaxError, ":").WithArguments("=>").WithLocation(3, 11),
+            // (3,15): error CS1041: Identifier expected; 'case' is a keyword
+            //     case 0: 1,
+            Diagnostic(ErrorCode.ERR_IdentifierExpectedKW, "").WithArguments("", "case").WithLocation(3, 15),
+            // (4,11): error CS1003: Syntax error, '=>' expected
+            //     case 1: 2,
+            Diagnostic(ErrorCode.ERR_SyntaxError, ":").WithArguments("=>").WithLocation(4, 11));
+        N(SyntaxKind.SwitchExpression);
+        {
+            N(SyntaxKind.IdentifierName);
+            {
+                N(SyntaxKind.IdentifierToken, "x");
+            }
+            N(SyntaxKind.SwitchKeyword);
+            N(SyntaxKind.OpenBraceToken);
+            N(SyntaxKind.SwitchExpressionArm);
+            {
+                N(SyntaxKind.ConstantPattern);
+                {
+                    N(SyntaxKind.NumericLiteralExpression);
+                    {
+                        N(SyntaxKind.NumericLiteralToken, "0");
+                    }
+                }
+                M(SyntaxKind.EqualsGreaterThanToken);
+                N(SyntaxKind.NumericLiteralExpression);
+                {
+                    N(SyntaxKind.NumericLiteralToken, "1");
+                }
+            }
+            N(SyntaxKind.CommaToken);
+            N(SyntaxKind.SwitchExpressionArm);
+            {
+                N(SyntaxKind.ConstantPattern);
+                {
+                    N(SyntaxKind.NumericLiteralExpression);
+                    {
+                        N(SyntaxKind.NumericLiteralToken, "1");
+                    }
+                }
+                M(SyntaxKind.EqualsGreaterThanToken);
+                N(SyntaxKind.NumericLiteralExpression);
+                {
+                    N(SyntaxKind.NumericLiteralToken, "2");
+                }
+            }
+            N(SyntaxKind.CommaToken);
+            N(SyntaxKind.CloseBraceToken);
+        }
+        EOF();
+        EOF();
+    }
+
+    [Fact]
+    public void TestErrantCaseInSwitchExpression2_Semicolons()
+    {
+        UsingExpression("""
+            x switch
+            {
+                case 0: 1;
```

Comment:
Can you add an example with multiple statements in a case?

Reply:
i can.  but it will not be good.  What this change helps me with though is as i'm mutating/adding things right around the case-clause.  in other words.  if there's a preexisting case, and i'm trying to update hte pattern that follows it.  Before this change, it's all non-functional.  After the change, i can make those changes, then remove the 'case'.  Similarly, i often have converted the case/pattern part, and i go to work on the arm, only to find things screwed up because of the `:`. reinterpreting it as a `=>` makes it easier to work with the rhs, then fixup the colon after :)

Reply:
@333fred test added.